src/arch/riscv/isa/decoder.isa

```diff
@@ -488,6 +488,15 @@ decode QUADRANT default Unknown::unknown() {
         format CompressedStore {
             0x5: decode ENABLE_ZCD {
                 0x0: decode CFUNCT6LOW3 {
+                    0x0: decode JTINDEX5TO7 {
```

Comment:
How are we going to deal with the fact that "This extension reuses some encodings from c.fsdsp"? 

Do the changes still work with original compressed extension?

Reply:
Hi @powerjg,

Yes.

We have provided the option `enable_Zcd` in `RiscvISA` to allow user to switch `c.fsdsp` and `Zcm*` instructions

https://github.com/gem5/gem5/blob/63d25922a2db14643f132c0f62a224c7f7991c49/src/arch/riscv/RiscvISA.py#L117-L123

We have `enable_zcd` checking in `decoder.isa`

https://github.com/gem5/gem5/blob/63d25922a2db14643f132c0f62a224c7f7991c49/src/arch/riscv/isa/decoder.isa#L489-L517

If `enable_zcd = 1`, we will decode it as `c.fsdsp`, otherwise decode it as `Zcm*` instructions.

Reply:
Got it. Thanks! I do think we should probably provide some error checking in the initialization of the RISC-V ISA object. I'm pretty sure that it's possible to configure it in a bad state right now. This can be done in another PR, though.